core/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -671,8 +649,12 @@ public ClusterState build() {
          * @param data      input bytes
          * @param localNode used to set the local node in the cluster state.
          */
-        public static ClusterState fromBytes(byte[] data, DiscoveryNode localNode) throws IOException {
-            return readFrom(StreamInput.wrap(data), localNode);
+        public static ClusterState fromBytes(byte[] data, DiscoveryNode localNode, NamedWriteableRegistry registry) throws IOException {
+            StreamInput input = StreamInput.wrap(data);
+            if (registry != null) {
```

Comment:
I'm not a fan of this null support either. I'd prefer sending an empty registry over null.